### ISSUE TITLE
Typescript definitios?

### ISSUE THREAD

**Issue by zmanian** [indutny/bn.js#157]
Hi, I've been slowly working my way through typescript definitions for the BN public API.

Would this be something you might be interested in merging in the future? The alternative is to submit them to DefinitelyTyped

**Comment by calvinmetcalf** [comment]
it's probably a better idea to put it into DefinitlyTyped unless one of the other maintainers thinks they can keep it up to date.  

In general I'm weary of merging a typescript definition into one of my projects because I don't use typescript so I can't maintain it, so it will never really be *official* and 100% sure to be up to date even through it's in the repo and looks to be official.

**Comment by dcousens** [comment]
Agreed with @calvinmetcalf ,  please see https://github.com/DefinitelyTyped/DefinitelyTyped

**Comment by EvanDarwin** [comment]
I'm interested in writing some TS declarations, I'll fork and submit a PR sometime...

**Comment by indutny** [comment]
👍thank you!

**Comment by ColtonProvias** [comment]
If it helps speed things up, I have this file from an old project of mine: https://gist.github.com/ColtonProvias/e9f1ebbfa9dcea4aa04d4e90bf0c42d5

**Comment by EvanDarwin** [comment]
Whoops I really let this go! 😢 (Turns out I didn't need bn.js)

**Comment by Levino** [comment]
Please see https://github.com/DefinitelyTyped/DefinitelyTyped/pull/26031

**Comment by krzkaczor** [comment]
If you want to use more natural import syntax use `ethereum-ts` package:
https://github.com/ethereum-ts/bn.js-typings
